File: sdk-python/platform_sdk/db/redis.py
import redis
import logging

logger = logging.getLogger('platform')


class RedisManager:
    def __init__(self, config: dict = None):
        import os as _os
        self.config = config or {
            "host": _os.environ.get("PLATFORM_REDIS_HOST", "localhost"),
            "port": int(_os.environ.get("PLATFORM_REDIS_PORT", "6379")),
            "password": _os.environ.get("PLATFORM_REDIS_PASSWORD", None),
        }
        self.client = None
        self.connected = False

    def connect(self):
        self.client = redis.Redis(**self.config, decode_responses=True)
        self.client.ping()
        self.connected = True
        logger.info("[platform] Redis connected")

    def get(self, key: str) -> str:
        if self.client:
            return self.client.get(key)
        return None

    def set(self, key: str, value: str, ttl: int = None):
        if self.client:
            if ttl:
                self.client.setex(key, ttl, value)
            else:
                self.client.set(key, value)

    def disconnect(self):
        if self.client:
            self.client.close()
            self.connected = False
            logger.info("[platform] Redis disconnected")

File: sdk-node/src/db/redis.ts
import Redis from 'ioredis';

export class RedisManager {
  private client: Redis | null = null;
  private connected = false;
  private retryCount = 0;
  private maxRetries = 5;
  private config: any;

  constructor(config: any = {}) {
    this.config = {
      host: config.host || process.env.PLATFORM_REDIS_HOST || 'localhost',
      port: config.port || parseInt(process.env.PLATFORM_REDIS_PORT || '6379', 10),
      password: config.password || process.env.PLATFORM_REDIS_PASSWORD || undefined,
    };
  }

  async connect(): Promise<void> {
    try {
      this.client = new Redis({ ...this.config, lazyConnect: true });
      await this.client.connect();
      this.connected = true;
      this.retryCount = 0;
      console.log('[platform] Redis connected');
    } catch (err: any) {
      console.error('[platform] Redis connection failed:', err.message);
      this.connected = false;
      this.scheduleReconnect();
    }
  }

  async get(key: string): Promise<string | null> {
    if (!this.client) return null;
    return this.client.get(key);
  }

  async set(key: string, value: string, ttl?: number): Promise<void> {
    if (!this.client) return;
    if (ttl) await this.client.setex(key, ttl, value);
    else await this.client.set(key, value);
  }

  async del(key: string): Promise<void> {
    if (!this.client) return;
    await this.client.del(key);
  }

  private scheduleReconnect(): void {
    if (this.retryCount >= this.maxRetries) return;
    const delay = Math.min(1000 * Math.pow(2, this.retryCount), 30000);
    this.retryCount++;
    setTimeout(() => this.connect(), delay);
  }

  async disconnect(): Promise<void> {
    if (this.client) {
      await this.client.quit();
      this.connected = false;
      console.log('[platform] Redis disconnected');
    }
  }

  get isConnected(): boolean { return this.connected; }
  get health(): { connected: boolean } {
    return { connected: this.connected };
  }
}
